src/openrct2/park/Legacy.cpp

```diff
@@ -2916,3 +2916,112 @@ bool TrackTypeMustBeMadeInvisible(ride_type_t rideType, OpenRCT2::TrackElemType
 
     return false;
 }
+
+OpenRCT2::ConvertedTrackTypeResult OpenRCT2::OldTrackElementToNew(OpenRCT2::OldTrackElemType trackElementType)
+{
+    switch (trackElementType)
+    {
+        case OldTrackElemType::FlatCovered:
+            return { TrackElemType::Flat, true };
+        case OldTrackElemType::Up25Covered:
+            return { TrackElemType::Up25, true };
+        case OldTrackElemType::Up60Covered:
+            return { TrackElemType::Up60, true };
+        case OldTrackElemType::FlatToUp25Covered:
+            return { TrackElemType::FlatToUp25, true };
+        case OldTrackElemType::Up25ToUp60Covered:
+            return { TrackElemType::Up25ToUp60, true };
+        case OldTrackElemType::Up60ToUp25Covered:
+            return { TrackElemType::Up60ToUp25, true };
+        case OldTrackElemType::Up25ToFlatCovered:
+            return { TrackElemType::Up25ToFlat, true };
+        case OldTrackElemType::Down25Covered:
+            return { TrackElemType::Down25, true };
+        case OldTrackElemType::Down60Covered:
+            return { TrackElemType::Down60, true };
+        case OldTrackElemType::FlatToDown25Covered:
+            return { TrackElemType::FlatToDown25, true };
+        case OldTrackElemType::Down25ToDown60Covered:
+            return { TrackElemType::Down25ToDown60, true };
+        case OldTrackElemType::Down60ToDown25Covered:
+            return { TrackElemType::Down60ToDown25, true };
+        case OldTrackElemType::Down25ToFlatCovered:
+            return { TrackElemType::Down25ToFlat, true };
+        case OldTrackElemType::LeftQuarterTurn5TilesCovered:
+            return { TrackElemType::LeftQuarterTurn5Tiles, true };
+        case OldTrackElemType::RightQuarterTurn5TilesCovered:
+            return { TrackElemType::RightQuarterTurn5Tiles, true };
+        case OldTrackElemType::SBendLeftCovered:
+            return { TrackElemType::SBendLeft, true };
+        case OldTrackElemType::SBendRightCovered:
+            return { TrackElemType::SBendRight, true };
+        case OldTrackElemType::LeftQuarterTurn3TilesCovered:
+            return { TrackElemType::LeftQuarterTurn3Tiles, true };
+        case OldTrackElemType::RightQuarterTurn3TilesCovered:
+            return { TrackElemType::RightQuarterTurn3Tiles, true };
+        default:
+        {
+            uint16_t asInt = EnumValue(trackElementType);
+            if (asInt >= EnumValue(OldTrackElemType::FirstCoveredElement))
+                asInt -= kNumLegacyCoveredElements;
+
+            return { static_cast<OpenRCT2::TrackElemType>(asInt), false };
+        }
+    }
```

Comment:
For good measure, both of these functions should also handle `::None`. (I don’t think it’s actually used in saves, but better be safe than sorry.)